cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```diff
@@ -48,12 +80,25 @@
      * @throws SQLException if the database cannot be opened.
      */
     public static SQLDatabase createSQLDatabase(String dbFilename, KeyProvider provider) throws IOException, SQLException {
+        makeSureFileExists(dbFilename);
+        return internalCreateSQLDatabase(dbFilename, provider);
+    }
+
+    /**
+     * Internal method for creating a SQLDatabase that allows a null filename to create an in-memory
+     * database which can be useful for performing checks, but creating in-memory databases is not
```

Comment:
Although our current usage is fine and it's private so it's only internally we'd be affected, this could either do with a comment stating that if `dbFilename` is null, `provider` must be a `NullKeyProvider`, or we need to update the `createAndroidSQLite` method in `AndroidSQLCipherSQLite` to handle creation of an encrypted in-memory database by calling `SQLiteDatabase.create` instead of `SQLiteDatabase.openOrCreateDatabase`.  As it stands, I think we'd get an `IllegalArgumentException` if it were called with a null `dbFilename` and a `provider` with a non-null key.